### PR TITLE
Filter ghost articles in export

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | yes
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT

#### What's in this PR?

This PR add the filter for ghost snippets in export.

### CODE REVIEW

File: Export/ArticleExport.php [hunk 1]
```diff
@@ -183,11 +183,25 @@ protected function getSettingData(ArticleDocument $document): array
 
     protected function getDocuments(string $locale): QueryResultCollection
     {
-        $query = $this->documentManager->createQuery(
-            'SELECT * FROM [nt:unstructured] AS a WHERE [jcr:mixinTypes] = "sulu:article"',
+        $where = [];
+
+        // only snippets
```
Comment (niklasnatter) [inline Export/ArticleExport.php]: 🚫

File: Export/ArticleExport.php [hunk 2]
```diff
@@ -183,11 +183,25 @@ protected function getSettingData(ArticleDocument $document): array
...
+        $where[] = '[jcr:mixinTypes] = "sulu:article"';
+
+        // filter by webspace key
```
Comment (niklasnatter) [inline Export/ArticleExport.php]: 🚫

**Comment by wachterjohannes** [comment]
@nnatter fixed

File: Export/ArticleExport.php [hunk 3]
```diff
@@ -183,11 +183,25 @@ protected function getSettingData(ArticleDocument $document): array
...
+
+        // filter by locale
+        $where[] = \sprintf(
+            '[i18n:%s-template] IS NOT NULL',
             $locale
         );
 
+        $query = $this->documentManager->createQuery(
+            'SELECT * FROM [nt:unstructured] AS a WHERE ' . \implode(' AND ', $where),
+            $this->exportLocale
```
Comment (niklasnatter) [inline Export/ArticleExport.php]: 🥴

File: Export/ArticleExport.php [hunk 4]
```diff
@@ -183,8 +183,22 @@ protected function getSettingData(ArticleDocument $document): array
...
     {
+        $where = [];
+
+        // only articles
+        $where[] = '[jcr:mixinTypes] = "sulu:article"';
+
+        // filter by path
+        $where[] = 'ISDESCENDANTNODE("/cmf/articles")';
+
+        // filter by locale
```
Comment (niklasnatter) [inline Export/ArticleExport.php]:
```suggestion
        // do not load ghost articles
```

**Pull request merged**